src/script_opt/ScriptOpt.cc

```diff
@@ -554,6 +554,9 @@ void clear_script_analysis()
 	{
 	IDOptInfo::ClearGlobalInitExprs();
 
+	// Keep the functions around if we're debugging, so we can
+	// generate profiles.
+#ifndef DEBUG
```

Comment:
Seems strange to make this conditional on `DEBUG`, shouldn't profiling work in release/default builds, too?

How about:
```
if ( ! analysis_options.profile_ZAM )
    ...    
```

---

Eh, okay. After trying and looking a bit, the ZAM profiling functionality is actually gated on `DEBUG` throughout, so I understand now where this is coming from.

Yet, leaving a note: A new option/define named `ZEEK_ZAM_PROFILING` might be worth considering. There may be users wanting to run Release builds and accepting a minor overhead (it's just a few more bool checks when available but not enabled it seems), but still want to have the option to use `-O profile-ZAM` without re-compiling.

`DEBUG` may also add extra overhead and skew results around a bit when profiling what should otherwise be an optimized build?

I was scratching my head a bit using `-O ZAM -O profile-ZAM` and it only showed "did not execute" when trying with a release build  - might be worth saying "ZAM profiling unavailable" so.

Anyway, merging :+1: